src/main/java/com/openshift/internal/restclient/model/Pod.java

```diff
@@ -61,10 +76,35 @@ public String getHost() {
 		}
 		return images;
 	}
+	
+	/**
+	 * The logic of the method is a copied from 'podStatus' function of
+	 * [app/scripts/filters/resources.js] of [openshift/origin-web-console]
+	 */
 
 	@Override
 	public String getStatus() {
-		return asString(POD_STATUS);
+		if (has(POD_DELETION_TIMESTAMP)) {
+			return "Terminating";
```

Comment:
@bdshadow that's exactly my point. I think that this should be in origin, not in tools (Eclipse or Web-UI). This makes me feel like they're lacking things if user code(s) have to add this state and  have extensive logic to determine a pod state (that's then replicated across these tools)